### PR TITLE
DTSPO-13563 - Add script and add pipeline

### PR DESCRIPTION
**Pull request by louisehuyton**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-13563

### Change description ###
Add script and add pipeline

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: scripts/check-toffee-plum.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    slack_message
+done
+}
+
+printf "\n:detective-pikachu: _*Check Toffee/Plum Status*_ \n">> slack-message.txt
+printf "\n*Toffee Status:*" >> slack-message.txt
+
+### test toffee
+APP="toffee"
+add_environments
```
Comment (timja) [inline scripts/check-toffee-plum.sh:47]: Do we maybe want to collapse the environments like in:
image

**Comment by louisehuyton** [comment]
Proof that multiple broken environments work:
image

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -197,12 +197,12 @@ jobs:
...
-        inputs:
-          targetType: filePath
-          filePath: scripts/jenkins-status.sh
-          arguments: $(jenkins-api-user) $(jenkins-api-token) ${{ variables.jenkins_url }}
+      # - task: Bash@3
+      #   displayName: 'Checking Jenkins status'
+      #   inputs:
+      #     targetType: filePath
+      #     filePath: scripts/jenkins-status.sh
+      #     arguments: $(jenkins-api-user) $(jenkins-api-token) ${{ variables.jenkins_url }}
```
Comment (timja) [inline azure-pipelines.yaml:205]: restore this?

File: scripts/check-toffee-plum.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+for ENV in ${ENVIRONMENTS[@]}; do
+    status_code $1
+    failure_check $1
+done
+}
+
+function do_failures_exist() {
+    if [[ $1 = "Toffee" ]]; then
+        if [[ $failures_exist_toffee != "true" ]]; then
+            printf "\n>:green_circle:  All environments in $1 are accessible" >> slack-message.txt
```
Comment (timja) [inline scripts/check-toffee-plum.sh:42]:
```suggestion
            printf "\n>:green_circle:  All environments in $1 are healthy" >> slack-message.txt
```

File: scripts/check-toffee-plum.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        fi
+    elif [[ $1 = "Plum" ]]; then
+        if [[ $failures_exist_plum != "true" ]]; then
+            printf "\n>:green_circle:  All environments in $1 are accessible" >> slack-message.txt
```
Comment (timja) [inline scripts/check-toffee-plum.sh:46]:
```suggestion
            printf "\n>:green_circle:  All environments in $1 are healthy" >> slack-message.txt
```

File: scripts/check-toffee-plum.sh [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    elif [ $ENV != "Prod" ]; then
+        url="https://$1.$ENV.platform.hmcts.net"
+        statuscode=$(curl -s -o /dev/null -w "%{http_code}" $url)
+    fi
+}
+
+function failure_check() {
+    if [[ $statuscode != 200 ]] && [[ $1 == "Toffee" ]]; then
+        failures_exist_toffee="true"
+        printf "\n>:red_circle:  <$url| $ENV>" >> slack-message.txt
```
Comment (timja) [inline scripts/check-toffee-plum.sh:25]:
```suggestion
        printf "\n>:red_circle:  <$url| $ENV> is unhealthy" >> slack-message.txt
```

File: scripts/check-toffee-plum.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    elif [[ $statuscode != 200 ]] && [[ $1 == "Plum" ]]; then
+        failures_exist_plum="true"
+        printf "\n>:red_circle:  <$url| $ENV>" >> slack-message.txt
```
Comment (timja) [inline scripts/check-toffee-plum.sh:28]:
```suggestion
        printf "\n>:red_circle:  <$url| $ENV> is unhealthy" >> slack-message.txt
```

File: azure-pipelines.yaml (hunk 2 above)
Comment (louisehuyton) [reply azure-pipelines.yaml:205]: yes apologies

**Review by timja — COMMENTED** [review]
looks good although the formatting of the script is all over the place would you be able to format it please?

this could do it automatically for you:
https://github.com/patrickvane/shfmt

**Review by timja — APPROVED** [review]
thanks looks much better

**Pull request merged**